contracts/asset-proxy/contracts/src/ERC20BridgeProxy.sol

```diff
@@ -0,0 +1,128 @@
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+pragma solidity ^0.5.9;
+pragma experimental ABIEncoderV2;
+
+import "@0x/contracts-utils/contracts/src/LibBytes.sol";
+import "@0x/contracts-utils/contracts/src/LibSafeMath.sol";
+import "@0x/contracts-utils/contracts/src/Authorizable.sol";
+import "@0x/contracts-erc20/contracts/src/interfaces/IERC20Token.sol";
+import "./interfaces/IAssetProxy.sol";
+import "./interfaces/IERC20Bridge.sol";
+
+
+contract ERC20BridgeProxy is
+    IAssetProxy,
+    Authorizable
+{
+    using LibBytes for bytes;
+    using LibSafeMath for uint256;
+
+    // @dev Result of a successful bridge call.
+    bytes4 constant public BRIDGE_SUCCESS = 0xb5d40d78;
+    // @dev Id of this proxy.
+    //      bytes4(keccak256("ERC20BridgeProxy(address,address,bytes)"))
+    bytes4 constant private PROXY_ID = 0x37708e9b;
+
+    /// @dev Calls a bridge contract to transfer `amount` of ERC20 from `from`
+    ///      to `to`. Asserts that the balance of `to` has increased by `amount`.
+    /// @param assetData Abi-encoded data for this asset proxy encoded as:
+    ///          abi.encodeWithSelector(
+    ///             bytes4 PROXY_ID,
+    ///             address tokenAddress,
+    ///             address bridgeAddress,
+    ///             bytes bridgeData
+    ///          )
+    /// @param from Address to transfer asset from.
+    /// @param to Address to transfer asset to.
+    /// @param amount Amount of asset to transfer.
+    function transferFrom(
+        bytes calldata assetData,
+        address from,
+        address to,
+        uint256 amount
+    )
+        external
+        onlyAuthorized
+    {
+        // Extract asset data fields.
+        (
+            address tokenAddress,
+            address bridgeAddress,
+            bytes memory bridgeData
+        ) = abi.decode(
+            assetData.sliceDestructive(4, assetData.length),
+            (address, address, bytes)
+        );
+
+        // Remember the balance of `to` before calling the bridge.
+        uint256 balanceBefore = balanceOf(tokenAddress, to);
+        // Call the bridge, who should transfer `amount` of `tokenAddress` to
+        // `to`.
+        bytes4 success = IERC20Bridge(bridgeAddress).transfer(
+            bridgeData,
+            tokenAddress,
+            from,
+            to,
+            amount
+        );
+        // Bridge must return the magic bytes to indicate success.
+        require(success == BRIDGE_SUCCESS, "BRIDGE_FAILED");
+        // Ensure that the balance of `to` has increased by at least `amount`.
+        require(
+            balanceBefore.safeAdd(amount) <= balanceOf(tokenAddress, to),
```

Comment:
Nit: I feel like `balanceOf(tokenAddress, to) >= balanceBefore.safeAdd(amount)` is slightly more intuitive. Not a huge deal either way.